ShowdownAI.Middleware/ShowdownAI.Middleware/Services/MoveDataLookup.cs

```diff
@@ -0,0 +1,103 @@
+﻿using ShowdownAI.Middleware.Models;
+using System.Text.Json;
+using System.Reflection;
+
+
+namespace ShowdownAI.Middleware.Services
+{
+    public class MoveDataLookup
+    {
+        private const string _moveDataPath = "./MoveData/move_data.json";
+        private readonly string _jsonData;    
+
+        public MoveDataLookup()
+        {
+            _jsonData = File.ReadAllText(_moveDataPath);
+        }
+
+        public MoveData GetMoveData(string moveId)
```

Comment:
It looks like every time we request a move it loops through every move in the json, is that right?
If so I suggest 2 improvements:

1. A Cache in this class, a dictionary<id, MoveData> which is populated whenever the move is found in the json and checked before the loop - Then exiting out of the for loop once the correct move has been found.
2. Creating a dictionary on startup which contains all the MoveData already populated so we're only reading the json once.

I don't know how fast it will read the json and I don't know the distribution of moves that are used, meaning if only about 20 moves ever really get used, the cache option is probably better. Otherwise I prefer the second option.

Reply:
Updated to create the dictionary on startup and it seems performant. If the other method seems better in future then it can be swapped.